homeassistant/components/mazda/climate.py

```diff
@@ -0,0 +1,221 @@
+PRESET_DEFROSTER_FRONT_AND_REAR = "Front and Rear Defroster"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the switch platform."""
+    entry_data = hass.data[DOMAIN][config_entry.entry_id]
+    client = entry_data[DATA_CLIENT]
+    coordinator = entry_data[DATA_COORDINATOR]
+    region = entry_data[DATA_REGION]
+
+    async_add_entities(
+        MazdaClimateEntity(client, coordinator, index, region)
+        for index, data in enumerate(coordinator.data)
+        if data["isElectric"]
+    )
+
+
+class MazdaClimateEntity(MazdaEntity, ClimateEntity):
+    """Class for the climate entity."""
+
+    _attr_name = "Climate"
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_hvac_modes = [HVACMode.HEAT_COOL, HVACMode.OFF]
+    _attr_preset_modes = [
+        PRESET_DEFROSTER_OFF,
+        PRESET_DEFROSTER_FRONT,
+        PRESET_DEFROSTER_REAR,
+        PRESET_DEFROSTER_FRONT_AND_REAR,
+    ]
+
+    _hvac_mode: HVACMode
+    _target_temperature: float
+    _preset_mode: str
+
+    def __init__(
+        self,
+        client: MazdaAPIClient,
+        coordinator: DataUpdateCoordinator,
+        index: int,
+        region: str,
+    ) -> None:
+        """Initialize Mazda climate entity."""
+        super().__init__(client, coordinator, index)
+
+        self.region = region
+        self._attr_unique_id = self.vin
+
+    async def async_added_to_hass(self) -> None:
+        """When entity is added to hass."""
+        # Whenever a coordinator update happens, update the state of the climate entity.
+        self.async_on_remove(
+            self.coordinator.async_add_listener(self._update_hvac_state)
+        )
+
+        # Perform an initial update of the state.
+        self._update_hvac_state()
+
+    def _update_hvac_state(self):
+        hvac_on = self.client.get_assumed_hvac_mode(self.vehicle_id)
+        self._hvac_mode = HVACMode.HEAT_COOL if hvac_on else HVACMode.OFF
+
+        hvac_setting = self.client.get_assumed_hvac_setting(self.vehicle_id)
+        self._target_temperature = hvac_setting.get("temperature")
+
+        front_defroster = hvac_setting.get("frontDefroster")
+        rear_defroster = hvac_setting.get("rearDefroster")
+        if front_defroster and rear_defroster:
+            self._preset_mode = PRESET_DEFROSTER_FRONT_AND_REAR
+        elif front_defroster:
+            self._preset_mode = PRESET_DEFROSTER_FRONT
+        elif rear_defroster:
+            self._preset_mode = PRESET_DEFROSTER_REAR
+        else:
+            self._preset_mode = PRESET_DEFROSTER_OFF
+
+        self.async_write_ha_state()
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return the current HVAC setting."""
+        return self._hvac_mode
+
+    @property
+    def precision(self) -> float:
```

Comment:
I think you can just set `self._attr_precision` in `__init__()`

Reply:
Fixed.